pkg/mqtrigger/scalermanager.go

```diff
@@ -0,0 +1,536 @@
+						"name":               mqt.ObjectMeta.Name,
+						"uid":                mqt.ObjectMeta.UID,
+						"blockOwnerDeletion": true,
+					},
+				},
+			},
+			"spec": map[string]interface{}{
+				"secretTargetRef": secretTargetRefFields,
+			},
+		},
+	}
+	return authTriggerObj, nil
+}
+
+func createAuthTrigger(mqt *fv1.MessageQueueTrigger, authenticationRef string, kubeClient *kubernetes.Clientset) error {
+	authTriggerObj, err := getAuthTriggerSpec(mqt, authenticationRef, kubeClient)
+	if err != nil {
+		return err
+	}
+	authTriggerClient, err := getAuthTriggerClient(mqt.ObjectMeta.Namespace)
+	if err != nil {
+		return err
+	}
+	_, err = authTriggerClient.Create(authTriggerObj, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func updateAuthTrigger(mqt *fv1.MessageQueueTrigger, authenticationRef string, kubeClient *kubernetes.Clientset) error {
+	authTriggerClient, err := getAuthTriggerClient(mqt.ObjectMeta.Namespace)
+	if err != nil {
+		return err
+	}
+	oldAuthTriggerObj, err := authTriggerClient.Get(authenticationRef, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	resourceVersion := oldAuthTriggerObj.GetResourceVersion()
+
+	authTriggerObj, err := getAuthTriggerSpec(mqt, authenticationRef, kubeClient)
+	if err != nil {
+		return err
+	}
+	authTriggerObj.SetResourceVersion(resourceVersion)
+	_, err = authTriggerClient.Update(authTriggerObj, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteAuthTrigger(name, namespace string) error {
+	authTriggerClient, err := getAuthTriggerClient(namespace)
+	if err != nil {
+		return err
+	}
+	err = authTriggerClient.Delete(name, &metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getDeploymentSpec(mqt *fv1.MessageQueueTrigger, routerURL string, kubeClient *kubernetes.Clientset) (*appsv1.Deployment, error) {
+	envVars, err := getEnvVarlist(mqt, routerURL, kubeClient)
+	if err != nil {
+		return nil, err
+	}
+	blockOwnerDeletion := true
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: mqt.ObjectMeta.Name,
+			Labels: map[string]string{
+				"app": mqt.ObjectMeta.Name,
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Kind:               "MessageQueueTrigger",
+					APIVersion:         "fission.io/v1",
+					Name:               mqt.ObjectMeta.Name,
+					UID:                mqt.ObjectMeta.UID,
+					BlockOwnerDeletion: &blockOwnerDeletion,
+				},
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": mqt.ObjectMeta.Name,
+				},
+			},
+			Template: apiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": mqt.ObjectMeta.Name,
+					},
+				},
+				Spec: apiv1.PodSpec{
+					Containers: []apiv1.Container{
+						{
+							Name:            mqt.ObjectMeta.Name,
+							Image:           "rahulbhati/test:10",
```

Comment:
@therahulbhati this should come from values.yaml - so for example for Kafka we have something like https://github.com/fission/fission/blob/master/charts/fission-all/values.yaml#L179 - similarly  we can have a new structure something like:

```
mq_keda:
  connector_images:
    kafka: fission/kafka-keda-connector:1.10.0
```

Reply:
Done